### PR TITLE
fix/types): all entry chain types

### PR DESCRIPTION
**Pull request by marcolink**
All Entry related types have been revised and adjusted. We have also added type definition tests for all of them.

**Required docs addition:**  
- `EntryFields.Link` should be used to define a linked entry 
- `EntryLink` defines an entry link in the response shape

**Todo:** 
Chek all types for Assets

### CODE REVIEW

**Comment by marcolink** [comment]
The PR is currently failing due to required fixes in the base PR #1102

File: package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
     "prettier": "prettier --write '**/*.{jsx,js,ts,tsx}'",
     "prettier:check": "prettier --check '**/*.{jsx,js,ts,tsx}'",
     "pretest": "rimraf coverage",
-    "test": "npm run test:unit && npm run test:integration && npm run lint",
+    "test": "npm run test:unit && npm run test:integration && npm run lint && npm run test:types",
```
Comment (damienxy) [inline package.json:53]: nice

File: test/types/entry-d.ts [hunk 2]
```diff
@@ -27,38 +37,275 @@ const entrySysValue: EntrySys = {
...
+        },
+      ],
+    },
+  },
+})
+
+/**
+ * @namespace: Typescript - type test
+ * @description: EntryWithLinkResolutionAndWithoutUnresolvableLinks only resolvable entries are present.
+ * unresolved entries are completely removed.
```
Comment (damienxy) [inline test/types/entry-d.ts:239]:
```suggestion
 * unresolvable entries are completely removed.
```

**Review by damienxy — APPROVED** [review]
Very nice

**Pull request merged**